File: picture_library/src/main/java/com/luck/picture/lib/PictureMediaScannerConnection.java
package com.luck.picture.lib;

import android.content.Context;
import android.media.MediaScannerConnection;
import android.net.Uri;
import android.text.TextUtils;

/**
 * @author：luck
 * @date：2019-12-03 10:41
 * @describe：刷新相册
 */
public class PictureMediaScannerConnection implements MediaScannerConnection.MediaScannerConnectionClient {
    public interface ScanListener {
        void onScanFinish();
    }

    private MediaScannerConnection mMs;
    private String mPath;
    private ScanListener mListener;

    public PictureMediaScannerConnection(Context context, String path, ScanListener l) {
        this.mListener = l;
        this.mPath = path;
        this.mMs = new MediaScannerConnection(context.getApplicationContext(), this);
        this.mMs.connect();
    }

    public PictureMediaScannerConnection(Context context, String path) {
        this.mPath = path;
        this.mMs = new MediaScannerConnection(context.getApplicationContext(), this);
        this.mMs.connect();
    }

    @Override
    public void onMediaScannerConnected() {
        if (!TextUtils.isEmpty(mPath)) {
            mMs.scanFile(mPath, null);
        }
    }

    @Override
    public void onScanCompleted(String path, Uri uri) {
        mMs.disconnect();
        if (mListener != null) {
            mListener.onScanFinish();
        }
    }
}

File: selector/src/main/java/com/luck/picture/lib/media/PictureMediaScannerConnection.kt
package com.luck.picture.lib.media

import android.content.Context
import android.media.MediaScannerConnection
import android.net.Uri
import android.text.TextUtils

/**
 * @author：luck
 * @date：2021/11/17 10:24 上午
 * @describe：Refresh System Album
 */
class PictureMediaScannerConnection(context: Context, path: String?, l: ScanListener?) :
    MediaScannerConnection.MediaScannerConnectionClient {

    private var mPath: String? = null
    private var mListener: ScanListener? = null
    private var mMs: MediaScannerConnection? = null

    init {
        this.mPath = path
        this.mListener = l
        if (path == null || TextUtils.isEmpty(path)) {
            mListener?.onScanFinish()
        } else {
            this.mMs = MediaScannerConnection(context.applicationContext, this)
            this.mMs?.connect()
        }
    }

    override fun onScanCompleted(path: String?, uri: Uri?) {
        mMs?.disconnect()
        mListener?.onScanFinish()
    }

    override fun onMediaScannerConnected() {
        if (!TextUtils.isEmpty(mPath)) {
            mMs?.scanFile(mPath, null)
        }
    }
}